### PR TITLE
[ONNX] Documentation for `torch.onnx.find_mismatch`

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #90728

Doc preview:
* `find_mismatch`: https://docs-preview.pytorch.org/90728/onnx.html#torch.onnx.verification.find_mismatch
* `GraphInfo`: https://docs-preview.pytorch.org/90728/onnx.html#classes and https://docs-preview.pytorch.org/90728/generated/torch.onnx.verification.GraphInfo.html#torch.onnx.verification.GraphInfo
* `VerificationOptions`: https://docs-preview.pytorch.org/90728/onnx.html#classes and  https://docs-preview.pytorch.org/90728/generated/torch.onnx.verification.VerificationOptions.html#torch.onnx.verification.VerificationOptions

### CODE REVIEW

File: torch/onnx/verification.py [hunk 1]
```diff
@@ -816,7 +819,7 @@ def verify(
     ] = None,
     input_names: Optional[Sequence[str]] = None,
     output_names: Optional[Sequence[str]] = None,
-    training: torch.onnx.TrainingMode = torch.onnx.TrainingMode.EVAL,
+    training: _C_onnx.TrainingMode = _C_onnx.TrainingMode.EVAL,
     opset_version: Optional[int] = None,
```
Comment (titaiwangms) [inline torch/onnx/verification.py:823]: Having the same concern from https://github.com/pytorch/pytorch/pull/91925#discussion_r1065238517
Do we also have a default in here?

**Review by titaiwangms — APPROVED** [review]

File: torch/onnx/verification.py (hunk 1 above)
Comment (justinchuby) [reply torch/onnx/verification.py:823]: Can we just make it required?

Comment (BowenBao) [reply torch/onnx/verification.py:823]: > Having the same concern from [#91925 (comment)](https://github.com/pytorch/pytorch/pull/91925#discussion_r1065238517) Do we also need a default here?

It is mimicking export api, which accepts None as input. I agree perhaps it's best to just add the default at top level and make everywhere else required. On the other PR I wasn't sure if the fx-onnx internals intercepts the None and assigns default value somewhere.

**Review by justinchuby — APPROVED** [review]
lgtm!

File: torch/onnx/verification.py [hunk 2]
```diff
@@ -1724,26 +1757,92 @@ def find_mismatch(
...
     verbose: bool = False,
     options: Optional[VerificationOptions] = None,
 ) -> GraphInfo:
     r"""Find all mismatches between the original model and the exported model.
-    TODO: Fill in docstring.
 
     Experimental. The API is subject to change.
 
+    This tool helps debug the mismatch between the original PyTorch model and exported
+    ONNX model. It binary searches the model graph to find the minimal subgraph that
```
Comment (justinchuby) [inline torch/onnx/verification.py:1771]: nit: runs binary search on

Comment (BowenBao) [inline torch/onnx/verification.py:1771]:
```suggestion
    ONNX model. It runs binary search on the model graph to find the minimal subgraph that
```

**Pull request closed without merging**